qutebrowser/completion/models/completion.py

```diff
@@ -215,6 +215,63 @@ def _init_settings(self):
                 self.new_item(cat, name, desc)
 
 
+class UrlCompletionModel(base.BaseCompletionModel):
+
+    """A CompletionModel which combines both quickmarks and web history
+    URLs. Used for the `open` command."""
+
+    def __init__(self, match_field='url', parent=None):
+        super().__init__(parent)
+
+        self._quickcat = self.new_category("Quickmarks")
+        self._histcat = self.new_category("History")
+        self._histstore = objreg.get('web-history')
+
+        WebHistoryCompletionModel.fill_model(self, cat=self._histcat)
+        QuickmarkCompletionModel.fill_model(self, cat=self._quickcat)
+
+        self._histstore.item_added.connect(lambda e:
+                                        WebHistoryCompletionModel.history_changed(
+                                            self, e, self._histcat))
```

Comment:
I'd prefer this to be written like this:

```python
        self._histstore.item_added.connect(
            lambda e: WebHistoryCompletionModel.history_changed(
                self, e, self._histcat))
```

But as said, I think calling `WebHistoryCompletionModel.history_changed` with `self` actually being an `UrlCompletionModel` is rather confusing - I'll check what the best way is to do this (normal functions/inherit `UrlCompletionModel` from `WebHistoryCompletionModel` and `QuickmarkCompletionModel`, have those models as members for `UrlCompletionModel`)